core/echo_test.go

```diff
@@ -102,3 +106,33 @@ func Test_getGroup(t *testing.T) {
 	assert.Equal(t, "", getGroup(""))
 	assert.Equal(t, "", getGroup("/"))
 }
+
+func TestHttpErrorHandler(t *testing.T) {
+	e := creatorFn().(*echo.Echo)
+	server := httptest.NewServer(e)
+	client := http.Client{}
+	t.Run("Problem return", func(t *testing.T) {
+		f := func(c echo.Context) error {
+			return problem.New(problem.Status(http.StatusInternalServerError))
+		}
+		e.Add(http.MethodGet, "/problem", f)
+		req, _ := http.NewRequest(http.MethodGet, fmt.Sprintf("%s/problem", server.URL), nil)
+		resp, err := client.Do(req)
+
+		assert.NoError(t, err)
+		assert.Equal(t, http.StatusInternalServerError, resp.StatusCode)
```

Comment:
Test some more. We expect a json struct on the correct content type right? There should be a title and description in here too?

Reply:
added test to check content type, and response body.